src/user/user.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import { hash, compare } from 'bcrypt';
+import { User } from '../user/entities/user.entity';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import * as moment from 'moment';
+
+interface UserInterface {
```

Comment:
(suggestion; non-blocking): We can probably replace this interface with `Partial<User>` instead.